readthedocs/oauth/tasks.py

```diff
@@ -134,9 +137,9 @@ def sync_active_users_remote_repositories():
     and it will require a pretty high ``time_limit`` and ``soft_time_limit``.
     """
     today_weekday = timezone.now().isoweekday()
-    three_months_ago = timezone.now() - datetime.timedelta(days=90)
+    one_month_ago = timezone.now() - datetime.timedelta(days=30)
     users = User.objects.annotate(weekday=ExtractIsoWeekDay("last_login")).filter(
-        last_login__gt=three_months_ago,
+        last_login__gt=one_month_ago,
```

Comment:
It feels like we shouldn't consider the user inactive until after longest possible user session duration, plus a buffer of some days. `last_login` is updated on log in, and so won't update again until the user's session is invalidated -- likely the session cookie aging out after 30 days. After that period, `last_login` doesn't really signal an inactive user until maybe 2 additional weeks have passed (44 days total) and they haven't logged in again.

So, maybe say 45 days is probably the minimum and 90 days does seem like a lot.

Reply:
With 90 days the task takes 1h 40min to complete (2.4K users), and with 45 days it takes 44min (1.3K users).

Reply:
Changed to 45 days.